databricks/koalas/frame.py

```diff
@@ -1461,14 +1455,16 @@ def transpose(self, limit: Optional[int] = 1000):
         1    float64
         dtype: object
         """
-        if limit is not None:
-            pdf = self.head(limit + 1)._to_internal_pandas()
-            if len(pdf) > limit:
+        max_compute_count = get_option("compute.max_rows")
+        if max_compute_count:
```

Comment:
It intentionally used `is not None` to make it clear `None` means unlimited case. Otherwise, empty list `[]` or other false-equivalent values could mean unlimited case as well.

It's rather preference matter but let's keep it consistent. Let's do ` if max_compute_count is not None:`